00X-ABI-Stable-Module-API.md

```diff
@@ -0,0 +1,377 @@
+NODE_EXTERN int napi_get_string_utf8(napi_env e, napi_value v,
+                                     char* buf, int bufsize);
+
+
+// Methods to coerce values
+// These APIs may execute user script
+NODE_EXTERN napi_value napi_coerce_to_object(napi_env e, napi_value v);
+NODE_EXTERN napi_value napi_coerce_to_string(napi_env e, napi_value v);
+
+
+// Methods to work with Objects
+NODE_EXTERN napi_value napi_get_prototype(napi_env e, napi_value object);
+NODE_EXTERN napi_propertyname napi_property_name(napi_env e,
+                                                 const char* utf8name);
+NODE_EXTERN napi_value napi_get_propertynames(napi_env e, napi_value object);
+NODE_EXTERN void napi_set_property(napi_env e, napi_value object,
+                                   napi_propertyname name, napi_value v);
+NODE_EXTERN bool napi_has_property(napi_env e, napi_value object,
+                                   napi_propertyname name);
+NODE_EXTERN napi_value napi_get_property(napi_env e, napi_value object,
+                                         napi_propertyname name);
+NODE_EXTERN void napi_set_element(napi_env e, napi_value object,
+                                  uint32_t i, napi_value v);
+NODE_EXTERN bool napi_has_element(napi_env e, napi_value object, uint32_t i);
+NODE_EXTERN napi_value napi_get_element(napi_env e,
+                                        napi_value object, uint32_t i);
+
+
+// Methods to work with Arrays
+NODE_EXTERN bool napi_is_array(napi_env e, napi_value v);
+NODE_EXTERN uint32_t napi_get_array_length(napi_env e, napi_value v);
+
+// Methods to compare values
+NODE_EXTERN bool napi_strict_equals(napi_env e, napi_value lhs, napi_value rhs);
+
+
+// Methods to work with Functions
+NODE_EXTERN void napi_set_function_name(napi_env e, napi_value func,
+                                        napi_propertyname napi_value);
+NODE_EXTERN napi_value napi_call_function(napi_env e, napi_value scope,
+                                          napi_value func,
+                                          int argc, napi_value* argv);
+NODE_EXTERN napi_value napi_new_instance(napi_env e, napi_value cons,
+                                         int argc, napi_value* argv);
+
+// Napi version of node::MakeCallback(...)
+NODE_EXTERN napi_value napi_make_callback(napi_env e, napi_value recv,
+                                          napi_value func,
+                                          int argc, napi_value* argv);
+
+
+// Methods to work with napi_callbacks
+NODE_EXTERN int napi_get_cb_args_length(napi_env e, napi_func_cb_info cbinfo);
+NODE_EXTERN void napi_get_cb_args(napi_env e, napi_func_cb_info cbinfo,
+                                  napi_value* buffer, size_t bufferlength);
+NODE_EXTERN napi_value napi_get_cb_this(napi_env e, napi_func_cb_info cbinfo);
+// V8 concept; see note in .cc file
+NODE_EXTERN napi_value napi_get_cb_holder(napi_env e, napi_func_cb_info cbinfo);
```

Comment:
Do other engines distinguish between this and holder? (EDIT: Or is that what "V8 concept" refers to?)